osu.Game/Screens/Select/FooterV2/FooterButtonV2.cs

```diff
@@ -0,0 +1,195 @@
+                    Colour = colourProvider.Background3,
+                    RelativeSizeAxes = Axes.Both
+                },
+
+                //For elements that should not be sheared.
+                new Container
+                {
+                    Anchor = Anchor.CentreLeft,
+                    Origin = Anchor.CentreLeft,
+                    Shear = -SHEAR,
+                    RelativeSizeAxes = Axes.Both,
+                    Children = new Drawable[]
+                    {
+                        TextContainer = new Container
+                        {
+                            Anchor = Anchor.TopCentre,
+                            Origin = Anchor.TopCentre,
+                            Y = 42,
+                            AutoSizeAxes = Axes.Both,
+                            Child = text = new OsuSpriteText
+                            {
+                                //figma design says the size is 16, but due to the issues with font sizes 19 matches better
+                                Font = OsuFont.TorusAlternate.With(size: 19),
+                                AlwaysPresent = true
+                            }
+                        },
+                        icon = new SpriteIcon
+                        {
+                            Y = 12,
+                            Size = new Vector2(20),
+                            Anchor = Anchor.TopCentre,
+                            Origin = Anchor.TopCentre
+                        },
+                    }
+                },
+                new Container
+                {
+                    // The X offset has to multiplied as such to account for the fact that we only want to offset by the distance from the CenterLeft point of the container
+                    // not the whole shear width
+                    Shear = -SHEAR,
+                    Anchor = Anchor.BottomCentre,
+                    Origin = Anchor.Centre,
+                    Y = -10,
+                    Size = new Vector2(120, 6),
+                    Masking = true,
+                    CornerRadius = 3,
+                    Child = bar = new Box
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                    }
+                }
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            Enabled.BindValueChanged(_ => updateDisplay(), true);
+            OverlayState.BindValueChanged(_ => updateDisplay());
+        }
+
+        public GlobalAction? Hotkey;
+
+        protected override bool OnHover(HoverEvent e)
+        {
+            updateDisplay();
+            return true;
+        }
+
+        protected override void OnHoverLost(HoverLostEvent e) => updateDisplay();
+
+        protected override bool OnMouseDown(MouseDownEvent e) => !Enabled.Value || base.OnMouseDown(e);
+        protected override bool OnClick(ClickEvent e) => !Enabled.Value || base.OnClick(e);
```

Comment:
I'm very confused by these conditionals. I do not think they should exist.

Reply:
what about them to be exact? i took this design pattern from the old footer buttons that were recently updated. although i did inline it prompted by the jetbrains tips, if thats the issue i can revert it, just making sure

Reply:
I'm not sure what the purpose is. Can you explain the purpose? If you can't, then it's probably not a good sign.

One rule to follow is to not blindly copy code, even if you are working with existing code. I believe these conditionals are going to make mouse clicks travel *through* the button to elements behind them when the buttons are not enabled, which is probably incorrect. I'd expect them to always be blocking mouse down / clicks?